### ISSUE TITLE
The .git folder should be excluded by default

### ISSUE THREAD

**Issue by alerque** [editorconfig-checker/action-editorconfig-checker#13]
I added a `[*]` rule to my `.editorconfig` files across a few projects with a `trim_trailing_whitespace = true` declaration, and now I'm getting CI failures across the board. Most of them are bogus. Specifically everything in `.git/objects/*` is being linted, and of course they don't match the lint style.

I think this should be excluded by default from the CI runner.

**Comment by alerque** [comment]
Come to think of it, probably `.gitignore` entries should probably be skipped by default. I'm getting false positive errors on more that just the .git directory.

**Comment by mstruebing** [comment]
I think that should be the case already, normally the tool only runs on git tracked files in case it's a git repository if you do not pass in specific files.
It uses the `git` tool directly to figure that out but I assume you have `git` available in your runner.

I tried to find the place in the action where it executes `editorconfig-checker` , it only downloads the binary as far as I can see but I may be wrong.

cc @septs do you have any idea what's the underlying issue?

**Comment by septs** [comment]
if you can provide the execution log?

**Comment by alerque** [comment]
Here is one: https://github.com/RichiH/vcsh/actions/runs/6261400486/job/17001414550

**Comment by septs** [comment]
https://github.com/septs/vcsh/actions/runs/6272503399/job/17034144780

`main` branch github action, tested OK

```diff
diff --git a/.github/workflows/lint.yml b/.github/workflows/lint.yml
index 24bb269e..459b2fa7 100644
--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -10,7 +10,8 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Lint code style
-        uses: editorconfig-checker/action-editorconfig-checker@v1.0.0
+        uses: editorconfig-checker/action-editorconfig-checker@main
+      - run: editorconfig-checker
 
   shellcheck:
     runs-on: ubuntu-latest
```

I think the problem with [v1.0.0](https://github.com/editorconfig-checker/action-editorconfig-checker/releases/tag/v1.0.0), but the version is unmaintained

**Comment by alerque** [comment]
Can you please tag a new `v2` release. I see there was a major refactor from Docker to typescript. If the current branch state is what you expect people to use that's fine, but please tag it to indicate that. There are other issues with unpinned actions downstream. There doesn't need to be any fancy release commit or changelog, just add a tag that indicates a version you expect to currently work.

**Comment by septs** [comment]
i not is the repo maintainer, only contributor

**Comment by alerque** [comment]
Fair enough. Thanks for contributing!

At least one project member is active in this thread through, so this request for a tag goes to to them ;-)

**Comment by mstruebing** [comment]
I will do - thank you guys for figuring this out 🙏 

> At least one project member is active in this thread through, so this request for a tag goes to to them ;-)

I'm always happy to add new maintainers if they are eager - I've maintained this project for so long that it's just not on my high-priority list anymore, so some things might take longer or need outside contributions to get in.
Let me know if you want to become an maintainer :)

**Comment by mstruebing** [comment]
@alerque could you please try this release: https://github.com/editorconfig-checker/action-editorconfig-checker/releases/tag/v2.0.0

I'll go ahead and close this issue. Feel free to reopen if this issue still persists.

**Comment by alerque** [comment]
Yes that worked, thank you!

Can you also add a `v2` rolling-release tag that tracks just the latest major version (so any `v.2.Y.Z` releases can still be tagged but `v2` will always point to the latest). This is recommended by the [GH Actions Versioning Guide](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md#versioning) and quite handy downstream to allow minor updates without having to add commit noise to downstream projects but also stay on the same major version to avoid potentially breaking changes.

**Comment by mstruebing** [comment]
Could you create a new issue for that please @alerque?
I think this would be good to work on once we make any additional changes.

**Comment by alerque** [comment]
I don't think it needs a tracking issue, it just needs a tag. You already had a `v1` tag for the `1.x.y` releases, and now you have a `2.0.0` release and a `v2` tag pointing to the same commit as the `v2.0.0` release is all that's needed. You can work out how you want to bump it later when you make more releases. I use [actions-tagger](https://github.com/Actions-R-Us/actions-tagger) on some projects to keep it fresh but manual bumps work fine too.

**Comment by mstruebing** [comment]
Done: https://github.com/editorconfig-checker/action-editorconfig-checker/releases/tag/v2

Not sure how to bump it yet but there will be ways.
